core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -522,10 +522,11 @@ private IndexShardState changeState(IndexShardState newState, String reason) {
         }
     }
 
-    public Engine.Index prepareIndexOnReplica(SourceToParse source, long seqNo, long version, VersionType versionType, long autoGeneratedIdTimestamp,
-                                              boolean isRetry) {
+    public Engine.Index prepareIndexOnReplica(SourceToParse source, long seqNo, long primaryTerm, long version, VersionType versionType,
```

Comment:
I spoke to Yannick on a different channel. We decided to rename the parameter name opPrimaryTerm and relax the assertion to reflect semantics (as opposed to current usage).